### PR TITLE
refactor(core): unify styling data format in attributes

### PR DESCRIPTION
**Pull request by AndrewKushnir**
Currently styling-related attributes (`class` and `style`) are represented differently depending on the node type.
Given the following template:
```
<div *ngIf="exp" title="abc" class="a b c" style="margin: 10px"></div>
```
For element node (that represent a `<div>`), this information is present in the parsed format (separated by
`AttributeMarker`s), such as:
```
["title", "abc", ${AttrMarker.Classes}, "a", "b", "c", ${AttrMarker.Styles}, "margin", "10px"]
```
When for the template node (that represents desugared version of structural directive, such as `<ng-template [ngIf]>`),
the class and style information would be present in the raw format:
```
["title", "abc", "class", "a b c", "style", "margin: 10px", ${AttrMarker.Template}, "ngIf"]
```
The same applies to the `<ng-content class="a b c">` tags (raw class information is present there).

Such inconsistency requires extra logic at runtime to perform matching and also blocks potential future work to
deduplicate attributes shared between element nodes and inline template nodes.

This commit updates the logic to represent class and style attrs info consistently (in parsed format) for all node
types.

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:345997704:BASE:345997717:1607306452107:abbfb072).

File: packages/core/src/render3/node_selector_matcher.ts [hunk 1]
```diff
@@ -24,27 +23,15 @@ const NG_TEMPLATE_SELECTOR = 'ng-template';
...
-  // and sometimes in form of `AttributeMarker.Classes`. Some investigation is needed to determine
-  // if that is the right behavior.
+function isCssClassMatching(attrs: TAttributes, cssClassToMatch: string): boolean {
   ngDevMode &&
       assertEqual(
           cssClassToMatch, cssClassToMatch.toLowerCase(), 'Class name expected to be lowercase.');
   let i = 0;
   while (i < attrs.length) {
     let item = attrs[i++];
-    if (isProjectionMode && item === 'class') {
```
Comment (petebacondarwin) [inline packages/core/src/render3/node_selector_matcher.ts:42]: Is the `isProjectionMode` not important information?
What was the scenario where the `class` attribute existed but it was not in projection mode?

**Comment by AndrewKushnir** [comment]
Quick update after discussion with @petebacondarwin: this work is a pre-requisite to the large refactoring to avoid generating duplicate attributes for inline templates (there is a portion of attributes that can be shared between "template" and "element" nodes). While this change seems to be useful as is, we'd like to create a followup PR that would implement actual deduplication, so that we can perform additional measurements to estimate the improvement to the payload size (and check if there are any runtime perf concerns). Keeping this PR in a draft state for now. Thank you.

**Comment by pkozlowski-opensource** [comment]
@AndrewKushnir it looks like the work on this PR stalled a bit so I'm going to close it for now. Feel free to re-open when you are ready to restart work on it.

**Pull request closed without merging**